src/include/migraphx/op/binary.hpp

```diff
@@ -13,10 +13,14 @@ struct binary : op_name<Derived>
     shape compute_shape(std::vector<shape> inputs) const
     {
         check_shapes{inputs}.has(2).same_type().same_dims();
-        const auto& s = inputs.front();
-        if(s.scalar() and s.elements() == 1)
-            return {s.type()};
-        return {s.type(), s.lens()};
+        if(inputs.at(0) == inputs.at(1) and inputs.at(0).packed())
```

Comment:
This should be different. If the shapes are the same we return `inputs.front()`. If the shapes are different return a standard shape.

Reply:
If the inputs shape is something like the sliced output, should we return a output shape as that from the slice operator?

Reply:
Thats a good point.